docker-compose.yml

```diff
@@ -15,732 +15,779 @@
+    entrypoint: *conda-entrypoint
+    command:
+      ["/arrow/ci/scripts/cpp_conda_build.sh /arrow /build/cpp &&
+        /arrow/ci/scripts/python_build.sh /arrow &&
+        /arrow/ci/scripts/integration_dask.sh"]
+
+  conda-python-turbodbc:
+    # document to use --no-cache in case of TURBODBC=master
+    image: ${ORG}/${ARCH}-conda-latest-python-${PYTHON}-turbodbc-${TURBODBC}:latest
+    build:
+      context: .
+      dockerfile: ci/docker/conda-latest-python-turbodbc.dockerfile
+      cache_from:
+        - ${ORG}/${ARCH}-conda-latest-python-${PYTHON}-turbodbc-${TURBODBC}:latest
+      args:
+        org: ${ORG}
+        arch: ${ARCH}
+        conda: latest
+        python: ${PYTHON}
+        turbodbc: ${TURBODBC}
+    shm_size: *shm-size
+    volumes: *conda-volumes
+    entrypoint: *conda-entrypoint
+    command:
+      ["/arrow/ci/scripts/cpp_conda_build.sh /arrow /build/cpp &&
+        /arrow/ci/scripts/python_build.sh /arrow &&
+        /arrow/ci/scripts/integration_turbodbc.sh /turbodbc /build/turbodbc"]
+
+  ########################## Python Wheels ####################################
+
+  centos-python-manylinux1:
+    image: ${ORG}/amd64-centos-python-manylinux1:latest
+    build:
+      context: python/manylinux1
+      dockerfile: Dockerfile-x86_64_base
+      cache_from:
+        - ${ORG}/amd64-centos-python-manylinux1:latest
+    shm_size: *shm-size
     environment:
-      PARQUET_TEST_DATA: /arrow/cpp/submodules/parquet-testing/data
-    volumes: *ubuntu-bionic-volumes
+      PYARROW_PARALLEL: 3
+      PYTHON_VERSION: ${PYTHON_VERSION:-3.6}
+      UNICODE_WIDTH: ${UNICODE_WIDTH:-16}
+    volumes: &manylinux-volumes
+      - .:/arrow:delegated
+      - ./python/manylinux1:/io:delegated
+    command: &manylinux-command /io/build_arrow.sh
 
-  go:
-    # Usage:
-    #   docker-compose build go
-    #   docker-compose run go
-    image: arrowdev/arrow-go:latest
+  centos-python-manylinux2010:
+    image: ${ORG}/amd64-centos-python-manylinux2010:latest
     build:
-      context: .
-      dockerfile: go/Dockerfile
-    volumes: *ubuntu-volumes
+      context: python/manylinux2010
+      dockerfile: Dockerfile-x86_64_base
+      cache_from:
+        - ${ORG}/amd64-centos-python-manylinux2010:latest
+    shm_size: *shm-size
+    environment:
+      PYARROW_PARALLEL: 3
+      PYTHON_VERSION: ${PYTHON_VERSION:-3.6}
+      UNICODE_WIDTH: ${UNICODE_WIDTH:-16}
+    volumes: *manylinux-volumes
+    command: *manylinux-command
 
-  integration:
-    image: arrowdev/arrow-integration:latest
-    build:
-      context: .
-      dockerfile: integration/Dockerfile
-    volumes:
-      - .:/arrow:ro  # ensures that docker won't contaminate the host directory
-      - maven-cache:/root/.m2:delegated
+  ################################## R ########################################
 
-  java:
-    # Usage:
-    #   docker-compose build java
-    #   docker-compose run java
-    image: arrowdev/arrow-java:latest
+  conda-r:
```

Comment:
@nealrichardson can we yank this, do we need to test conda and r?

Reply:
IMO the conda R job can be nightly (and furthermore should test the conda recipe; there is a Jira for this). It's not worth consuming a concurrent test runner for `conda-r`.

Reply:
Reading in the context of the diff, it may make sense for conda-r to be in docker-compose.yml, but it shouldn't be wired up in github actions.

Reply:
This is quite important, it runs the protocol/format tests, so we should definitely keep it.